### PR TITLE
Add Sentry for exception monitoring

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

We want to prepare for TIMDEX-UI as a potential production
application.

#### Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/TIMX-81

#### How this addresses that need:

This adds sentry-ruby and sentry-rails, including the required
initializer.

#### Side effects of this change:

* Configures Sentry environment. We haven't done this yet in our
Rails apps, but it seems like it will be especially useful for
TIMDEX and TIMDEX-UI, for which we have a lot of different
environments.
* Added SENTRY_ENV and SENTRY_DSN to Heroku config vars.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

YES

### CODE REVIEW

**Comment by jazairi** [comment]
Here's a test message I sent to Sentry via console: https://sentry.io/organizations/mit-libraries/issues/3492657166/?project=6638186&query=is%3Aunresolved. Note that the environment is 'pr'.

File: config/initializers/sentry.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Sentry.init do |config|
+  return unless ENV.has_key?('SENTRY_DSN')
+  config.dsn = ENV.fetch('SENTRY_DSN')
+  config.breadcrumbs_logger = [:active_support_logger, :http_logger]
+  config.environment = ENV.fetch('SENTRY_ENV', 'prod')
```
Comment (JPrevost) [inline config/initializers/sentry.rb:5]: This is great. I think we should move to this as our standard practice.

File: README.md [hunk 2]
```diff
@@ -53,6 +53,8 @@ change as part of the work.
 `basic_search#index`.
 - `GLOBAL_ALERT`: The main functionality for this comes from our theme gem, but when set the value will be rendered as
   safe html above the main header of the site.
+- `SENTRY_DSN`: Client key for Sentry exception logging.
+- `SENTRY_ENV`: Sentry environment for the application. Defaults to 'prod' if unset.
```
Comment (JPrevost) [inline README.md:57]: I wonder if we should have the default be something like `unknown` to clarify that we don't actually _know_ if it's in prod unless it's configured. I can see either side, but feel like it's worth considering.

**Review by JPrevost — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jazairi) [reply README.md:57]: Yeah, I thought about `unknown` as well. If you're good with that, I'll go ahead and make that change.

**Review by JPrevost — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (JPrevost) [reply README.md:57]: Looks like we updated the method but not the docs on that last change

**Review by JPrevost — CHANGES_REQUESTED** [review]
Docs should update to match the change in the method, but then :shipit:

**Comment by JPrevost** [comment]
I've created https://mitlibraries.atlassian.net/browse/ENGX-181 to add the environment to our other rails apps.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**